.github/workflows/ci.yml

```diff
@@ -112,16 +122,30 @@ jobs:
     # This prevents duplicate CI runs for our own pull requests, whilst preserving the ability to
     # run the CI for each branch push to a fork, and for each pull request originating from a fork.
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
-    runs-on: windows-latest
+    needs: setup-os-matrix
+    strategy:
+      matrix:
+        os: ${{ fromJson(needs.setup-os-matrix.outputs.os) }}
+        library: [ILGPU, ILGPU.Algorithms]
+        framework: [netcoreapp3.1, net5.0]
+        include:
+          - os: windows-latest
+            library: [ILGPU, ILGPU.Algorithms]
+            framework: net47
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
```

Comment:
that was my next question since build and package are confusing me (I originally copied the whole CI structure for Build and Package) 
Thanks for this gold input! 🙏 
Let me know if some change are still needed, with current CI we have [package ](https://github.com/ljubon/ILGPU/runs/3032061028?check_suite_focus=true)🟢 for all librarrys/framerwosk

Reply:
as you'll see below, you should essentially leave the `package` job as it is in the current CI, _except_ that we want to setup the latest .NET 5 SDK instead of .NET Core 3.1